### PR TITLE
Adding lesson 3 files

### PR DESCRIPTION
**Pull request by jineehan**
Adding (4) lesson 3 files: list_lab.py, mail.py, slicing_lab.py, and strformat_lab.py.

### CODE REVIEW

**Comment by jineehan** [comment]
Re-opening. Trying to better manage files (adding one file per pull request). Please only grade mail.py from the PR. I will create new PRs for the other scripts.

**Comment by natasha-aleksandrova** [comment]
@jineehan This PR includes Lesson 3: slicing, list lab, str lab and mailroom - are you sure you only want the mailroom reviewed? the rest look complete.

File: students/jinee_han/lesson03/Mail.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+# Display options
+prompt = "\n".join(("Welcome to the donor list!",
+          "Please choose from below options:",
+          "1 - Send a Thank you",
+          "2 - Create a Report",
+          "3 - Exit",
+          ">>> "))
+
+def send_thank_you_note():
+    inputValue = input("Enter a full name. (Type 'list' to see the donor list)")  # Ask the name
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/Mail.py]: variable names follow snake_case convention -> `input_value`

File: students/jinee_han/lesson03/Mail.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    while inputValue == 'list':
+        display_list()
+        inputValue = input("Enter a full name. (Type 'list' to see the donor list)")
+
+    if not donor_present(inputValue):
+        add_donor(inputValue)
+
+    donationName = inputValue
+    donationAmount = float(input("Please enter the donation amount: "))
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/Mail.py]: more pep8 violations (variable naming)

File: students/jinee_han/lesson03/Mail.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            currentDonationList = donator[1]
+            currentDonationList.append(donationAmount)
+            donator[1] = currentDonationList
+            break
+
+def donor_present(donorName):
+    donorPresent = False
+    for donor in donor_db:
+        if donorName == donor[0]:
+            donorPresent = True
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/Mail.py]: you can avoid using temp vars and just do `return True` here and `return False` on line 71

File: students/jinee_han/lesson03/Mail.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+def add_donor(donorName):
+    donationHistory = []
+    newDonator = [donorName, donationHistory]
+    donor_db.append(newDonator)
+
+def enter_amount():
+    input_amount = ("What is amount of the donation? ")
+    return ("You entered "+input_amount)
+    thankyou(input_list,input_amount)
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/Mail.py]: function call after return statement?

File: students/jinee_han/lesson03/Mail.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        average = total/num
+        print('{:^18} ${:>12,.2f}{:^13}  ${:>12,.2f}'.format(entry[0],total,num,average))
+    print('')
+
+def sort_key(entry):
+    return sum(entry[1])
+
+def exit_program():
+    print("Thank you for your donations!")
+    sys.exit()  # exit the interactive script
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/Mail.py]: 👍

File: students/jinee_han/lesson03/Slicing_Lab.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+# 1. First and last items exchanged.
+
+def exchange_first_last(seq):
+    new_line = seq[-1:] + seq[1:-1] + seq[:1]
+    return new_line
+
+# 2. Every other item removed.
+
+def every_other_letter_removed(seq):
+    new_line = seq[0:-1:2]
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/Slicing_Lab.py]: note that `0` is default so you can omit it

File: students/jinee_han/lesson03/List_Lab.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+
+# Display all the fruits that begins with 'P'
+
+print ("\nLet's find out fruits start with P\n")
+for items in fruit_list:
+    if items.startswith('P'):
+        print(items)
+    else:
+        continue
```
Comment (natasha-aleksandrova) [inline students/jinee_han/lesson03/List_Lab.py]: no need for the `else` and `continue` statements.

**Comment by jineehan** [comment]
> @jineehan This PR includes Lesson 3: slicing, list lab, str lab and mailroom - are you sure you only want the mailroom reviewed? the rest look complete.

@natasha-aleksandrova Please review all the files. I thought I should update them separately. It is great if I don't have to. Thank you!

File: students/jinee_han/lesson03/Mail.py (hunk 1 above)
Comment (jineehan) [reply students/jinee_han/lesson03/Mail.py]: Thank you. I will do that in the future.

File: students/jinee_han/lesson03/Mail.py (hunk 2 above)
Comment (jineehan) [reply students/jinee_han/lesson03/Mail.py]: Noted, thank you.

File: students/jinee_han/lesson03/Mail.py (hunk 3 above)
Comment (jineehan) [reply students/jinee_han/lesson03/Mail.py]: Thank you

File: students/jinee_han/lesson03/List_Lab.py (hunk 7 above)
Comment (jineehan) [reply students/jinee_han/lesson03/List_Lab.py]: Thank you

File: students/jinee_han/lesson03/Mail.py (hunk 4 above)
Comment (jineehan) [reply students/jinee_han/lesson03/Mail.py]: I meant to remove that function call. Thanks for catching it.

**Pull request merged**